HotTips/Engine/VSTipHistoryManager.cs

```diff
@@ -60,27 +72,41 @@ private string GetTipHistoryFromSettingsStore()
             return tipHistoryRaw;
         }
 
-        public void MarkTipAsSeen(string globalTipId)
+        public void MarkTipAsSeen(TipHistoryInfo globalTipId)
         {
             // Back out if tip already seen
-            List<string> tipsSeen = GetTipHistory();
-            if (tipsSeen.Contains(globalTipId))
+            List<TipHistoryInfo> tipsSeen = GetTipHistory();
+            if (tipsSeen.Exists(a => a.globalTipId.Equals(globalTipId.globalTipId)))
             {
                 // Item is already in the history. No need to add it again.
                 return;
             }
-
-            // Add tip ID to the tip history and persist to the VS settings store
-            tipsSeen.Add(globalTipId);
+            else
+            {
+                // Add tip ID to the tip history and persist to the VS settings store
+                tipsSeen.Add(globalTipId);
+            }
 
             // Update the VS settings store with the latest tip history
-            string tipHistoryRaw = String.Join(",", tipsSeen);
+            string tipHistoryRaw = String.Join(";", tipsSeen);
             // Get Writable settings store and update value
             string collectionPath = TIP_OF_THE_DAY_SETTINGS;
             //ISettingsList settingsList = SettingsManager.GetOrCreateList(collectionPath, isMachineLocal: !RoamSettings);
             SettingsManager.SetValueAsync(collectionPath, tipHistoryRaw, isMachineLocal: !RoamSettings);
         }
 
+        public void SaveTipStatus(TipHistoryInfo globalTipId)
```

Comment:
Does this method actually "SaveTipStatus"? It looks like it only adds it to the in-memory object, but does not persist it to the registry.